### PR TITLE
feat: Add custom header in KDrawer

### PR DESCRIPTION
**Pull request by jobsofferings**
Add custom header in KDrawer, and remove the previous footer slot, part of the UI library seems to want to combine drawer with forms, so I'm going to follow this.

### CODE REVIEW

File: components/Drawer/__test__/drawer.spec.ts [hunk 1]
```diff
@@ -84,13 +84,15 @@ describe('Test: KDrawer', () => {
 		expect(mockFn).toBeCalled();
 	});
 
-	test('slot: content', async () => {
-		const instance = new KDrawerContent({
-			target: host
+	test('slot: header', async () => {
```
Comment (baiwusanyu-c) [inline components/Drawer/__test__/drawer.spec.ts:87]: I still don't understand why you removed the slot related tests for content, can you explain?

**Review by baiwusanyu-c — CHANGES_REQUESTED** [review]

File: components/Drawer/__test__/drawer.spec.ts (hunk 1 above)
Comment (jobsofferings) [reply components/Drawer/__test__/drawer.spec.ts:87]: wait...i'll fix it

Comment (jobsofferings) [reply components/Drawer/__test__/drawer.spec.ts:97]: I know why I blocked this... I've rolled back this content and added snap

**Comment by baiwusanyu-c** [comment]
Do you have any other changes to commit?

**Comment by jobsofferings** [comment]
Not yet?

**Pull request merged**